include/deal.II/numerics/matrix_creator.templates.h

```diff
@@ -1332,94 +1335,97 @@ namespace MatrixCreator
-
-                                copy_data.cell_vector.back()(i) += rhs_values_system[point](
-                                                                     component_mapping[fe.system_to_component_index(i).first])
-                                                                   * weight
-                                                                   * v;
-                              }
-                          }
-                      }
-                    else
-                      {
-                        coefficient_vector_values.resize (fe_values.n_quadrature_points,
-                                                          Vector<number>(n_components));
-                        coefficient->vector_value_list (fe_values.get_quadrature_points(),
-                                                        coefficient_vector_values);
-                        for (unsigned int point=0; point<fe_values.n_quadrature_points; ++point)
-                          {
-                            const double weight = fe_values.JxW(point);
-                            for (unsigned int i=0; i<fe_values.dofs_per_cell; ++i)
-                              {
-                                const double v = fe_values.shape_value(i,point);
-                                const unsigned int component_i=
-                                  fe.system_to_component_index(i).first;
-                                for (unsigned int j=0; j<fe_values.dofs_per_cell; ++j)
-                                  if (fe.system_to_component_index(j).first ==
-                                      component_i)
-                                    {
-                                      const double u = fe_values.shape_value(j,point);
-                                      copy_data.cell_matrix.back()(i,j) +=
-                                        (coefficient_vector_values[point](component_i) * u * v * weight);
-                                    }
-                                copy_data.cell_vector.back()(i) += rhs_values_system[point](component_mapping[component_i])
-                                                                   * v
-                                                                   * weight;
-                              }
-                          }
-                      }
+                    const FiniteElement<dim,spacedim> &base = fe.base_element(fe.component_to_base_index(comp).first);
+                    const unsigned int bcomp = fe.component_to_base_index(comp).second;
+
+                    if (!base.conforms(FiniteElementData<dim>::H1) &&
+                        base.conforms(FiniteElementData<dim>::Hdiv))
+                      for (unsigned int point=0; point<fe_values.n_quadrature_points; ++point)
+                        normal_adjustment[point][comp] = fe_values.normal_vector(point)[bcomp]
+                                                         * fe_values.normal_vector(point)[bcomp];
```

Comment:
The documentation further up promises to also make the adjustment for Hcurl elements, but I don't actually see this here.

Reply:
I was using the non-hp `create_boundary_mass_matrix_1` function as a reference when doing these changes and it did not have it. I agree that the adjustment for the Hcurl should be done, and I think I now have a piece of code that does it, would you want me to add it to the non-hp function as well?

Reply:
In that case, can you just do something like `Assert (...., ExcNotImplemented())` where `...` is the appropriate condition to catch the Hcurl case? Bonus points if you do the same in the existing function!